tests/test_config.py

```diff
@@ -38,10 +38,10 @@ def _compare_configs(self, dict_config, server_config):
         subclass thereof.
 
     """
-    cfg = vars(server_config).copy()
-    if 'version' in cfg:
-        cfg['version'] = str(cfg['version'])
-    self.assertEqual(dict_config, cfg)
+    if 'version' in dict_config:
+        dict_config = dict_config.copy()  # shadow the passed in dict
```

Comment:
What I'm doing here is making a new variable called `dict_config` that is completely separate from the passed-in `dict_config`. Maybe this example will explain things a bit better:

```python
>>> # This is baaaaad.
>>> def do_not_shadow(arg):
...     arg[1] = 5
...
>>> arg = {1: 2}
>>> arg
{1: 2}
>>> do_not_shadow(arg)
>>> arg
{1: 5}
>>> # This is good!
>>> def do_shadow(arg):
...     arg = arg.copy()
...     arg[1] = 5
...
>>> arg = {1: 2}
>>> do_shadow(arg)
>>> arg
{1: 2}
```